internal/tasks/publishertask.go

```diff
@@ -0,0 +1,152 @@
+package tasks
+
+import (
+	"context"
+
+	"github.com/razorpay/metro/internal/common"
+	"github.com/razorpay/metro/internal/nodebinding"
+	"github.com/razorpay/metro/internal/topic"
+	"github.com/razorpay/metro/pkg/logger"
+	"github.com/razorpay/metro/pkg/registry"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// PublisherTask implements the Watcher and maintains a pre-warmup.
+type PublisherTask struct {
+	id              string
+	registry        registry.IRegistry
+	topicCore       topic.ICore
+	nodeBindingCore nodebinding.ICore
+	topicWatchData  chan *struct{}
+}
+
+var topicCacheData map[string]bool = make(map[string]bool)
```

Comment:
we should avoid having global variables, is there a way we can remove this..?

Reply:
@sakshi-goyal-razorpay even I was looking for better alternatives and suggestions to define a map inside a package.

Reply:
why can't we keep it as part of the task model itself just like topicCache in SchedulerTask..?